docs/userguides/testing.md

```diff
@@ -0,0 +1,178 @@
+# Testing
+
+Testing an ape project is important and easy.
+All tests must be stored under `tests/`. Each test must start with `test_` and end with the `.py` extension.
+
+### Test Structure
+
+Tests must be located in a project's `tests/` directory. Each test file must start with `test_` and have the `.py`
+extension, such as `test_my_contract.py`. Each test method within the file must also start with `test_`. The following
+is an example test:
+
+```python
+def test_add():
+    assert 1 + 1 == 2
+```
+### Test Pattern
+
+Tests are generally divisible into three parts:
+
+1. Set-up
+2. Invocation
+3. Assertion
+
+In the example above, we created a fixture that deploys our smart-contract. This is an example of a 'setup' phase.
+Next, we need to call a method on our contract. Let's assume there is a method called `is_owner()` that returns `True`
+when it is the owner of the contract making the transaction.
+
+This is an example of how that test may look:
+
+```python
+def test_is_owner(my_contract, owner, other):
+    owner_is_owner = my_contract.foo(sender=owner)
+    assert owner_is_owner
```

Comment:
Not sure what this contract looks like, but sending a transaction (via `sender=...`) means that `owner_is_owner` is actually a transaction receipt object, not a return value

I would think it would look something like this:
```suggestion
    my_contract.set_owner(sender=owner)
    assert owner == my_contract.owner()
```